lib/bloc/backup/backup_bloc.dart

```diff
@@ -0,0 +1,127 @@
+
+  final _multipleRestoreController = new StreamController<Map<String, String>>.broadcast();
+  Stream<Map<String, String>> get multipleRestoreStream => _multipleRestoreController.stream;
+
+  static const String BACKUP_PROMPT_DISABLED_PREFERENCES_KEY = "backupDisabled";
+
+  BackupBloc(this._accountBloc) {
+    ServiceInjector injector = new ServiceInjector();
+    BreezBridge breezLib = injector.breezBridge;
+    _service = BackupService();
+
+    var sharedPrefrences = SharedPreferences.getInstance();
+
+    _listenBackupPaths(breezLib, sharedPrefrences);
+    _listenEnableBackupRequests(sharedPrefrences);
+    _listenBackupNowRequests(sharedPrefrences);
+    _listenRestoreRequests(breezLib);
+
+    _accountBloc.accountStream.listen((acc) {
+      _currentNodeId = acc.id;
+    });
+  }
+
+  void _listenEnableBackupRequests(Future<SharedPreferences> sharedPrefrences) {
+    _disableBackupPromptController.stream.listen((disable) {
+      sharedPrefrences.then((preferences) {
+        preferences.setBool(BACKUP_PROMPT_DISABLED_PREFERENCES_KEY, disable);
+      });
+    });
+  }
+
+  void _listenBackupNowRequests(Future<SharedPreferences> sharedPrefrences) {
+    _backupNowController.stream.listen((data) {
+      backup();
+    });
+  }
+
+  _listenBackupPaths(BreezBridge breezLib, Future<SharedPreferences> sharedPrefrences) {
+    Observable(breezLib.notificationStream).where((event) {
+      return event.type ==
+          NotificationEvent_NotificationType.BACKUP_FILES_AVAILABLE;
+    }).listen((event) {
+      _currentBackupPaths = event.data;
+      sharedPrefrences.then((preferences) {
+        if (preferences.getBool(BACKUP_PROMPT_DISABLED_PREFERENCES_KEY) ?? false) {
+          // Prompt is disabled so go and back up
+          backup();
+        }
+        else {
+          // Prompting is enabled so show the dialog
+          _promptEnableBackupController.add(true);
+        }
+      });
+    });
+  }
+
+  void _listenRestoreRequests(BreezBridge breezLib) {
+    _restoreRequestController.stream.listen((nodeId) {
+      _service.restore(nodeId: nodeId).then((restoreResult) {
+        if (restoreResult is List<dynamic>) {
+          // We got a list of local files to restore from
+          // So let's kick-off lighntinglib
+          breezLib.bootstrap().then((done){
+            getApplicationDocumentsDirectory().then((appDir) {
+              breezLib.copyBreezConfig(appDir.path).then((done) {
+                breezLib.bootstrapFiles(appDir.path, new List<String>.from(restoreResult)).then((done) {
+                  _accountBloc.startLightningSink.add(true);
+                });
+              });
+            });
+          });
+        }
+        else {
+          // We need to pick from different node IDs
+          _multipleRestoreController.add(new Map<String, String>.from(restoreResult));
+        }
+      }).catchError((error) {
+        // Notify user somehow?
```

Comment:
perhaps we should add error to one of the streams and show error in the restore dialog UI?